pkg/agent/agent_test.go

```diff
@@ -708,3 +712,263 @@ func TestRestorePortConfigs(t *testing.T) {
 		})
 	}
 }
+
+func TestSetOVSDatapath(t *testing.T) {
+	tests := []struct {
+		name          string
+		expectedCalls func(m *ovsconfigtest.MockOVSBridgeClient)
+		expectedErr   string
+	}{
+		{
+			name: "fail to read OVS bridge other_config",
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().GetOVSOtherConfig().Return(nil, ovsconfig.InvalidArgumentsError("failed to read OVS bridge other_config"))
+			},
+			expectedErr: "failed to read OVS bridge other_config",
+		},
+		{
+			name: "fail to set OVS bridge datapath_id",
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().GetOVSOtherConfig().Return(map[string]string{}, nil)
+				m.EXPECT().SetDatapathID(mock.Any()).Return(ovsconfig.InvalidArgumentsError("failed to set OVS bridge datapath_id"))
+			},
+			expectedErr: "failed to set OVS bridge datapath_id",
+		},
+		{
+			name: "datapath-id exists in other_config on OVS bridge",
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().GetOVSOtherConfig().Return(map[string]string{
+					ovsconfig.OVSOtherConfigDatapathIDKey: "datapathId",
+				}, nil)
+			},
+		},
+		{
+			name: "generate and set datapath ID for OVS bridge",
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().GetOVSOtherConfig().Return(map[string]string{}, nil)
+				m.EXPECT().SetDatapathID(mock.Any())
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			controller := mock.NewController(t)
+			mockOVSBridgeClient := ovsconfigtest.NewMockOVSBridgeClient(controller)
+			initializer := newAgentInitializer(mockOVSBridgeClient, nil)
+			tt.expectedCalls(mockOVSBridgeClient)
+
+			err := initializer.setOVSDatapath()
+			if tt.expectedErr != "" {
+				assert.ErrorContains(t, err, tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func mockIPsecPSKEnv(name string) func() {
+	_ = os.Setenv(ipsecPSKEnvKey, name)
```

Comment:
No strong opinion, but suggest to call `os.Setenv(ipsecPSKEnvKey, name)` directly without `_ =`.